File: gg/gateways/git/cherry_pick.py
from plumbum import FG

from gg.gateways.git import git_cmd

def git_cherry_pick(start_ref: str, end_ref: str):
    """Cherry pick a series of changes onto the current checkout"""
    cmd = git_cmd['cherry-pick', '%s..%s' % (start_ref, end_ref)]

    (cmd) & FG

def git_cherry_pick_continue():
    """Cherry pick --continue"""
    cmd = git_cmd['cherry-pick', '--continue']

    (cmd) & FG

def git_cherry_pick_abort():
    """Cherry pick --continue"""
    cmd = git_cmd['cherry-pick', '--abort']

    (cmd) & FG

File: go/src/github.com/willhug/gg/git/cherrypick.go
package git

func (g *Git) CherryPick(startRef, endRef string) error {
	return g.runner.Run("cherry-pick", startRef+".."+endRef)
}

func (g *Git) CherryPickWithStrategy(startRef, endRef, strategy string) error {
	return g.runner.Run("cherry-pick", startRef+".."+endRef, "--strategy-option", strategy)
}

func (g *Git) CherryPickContinue() error {
	return g.runner.Run("cherry-pick", "--continue")
}

func (g *Git) CherryPickAbort() error {
	return g.runner.Run("cherry-pick", "--abort")
}
